File: algorithm/src/main/java/baekjoon/p02506/Main.java
package baekjoon.p02506;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] inputs = new int[n];

        for (int i = 0; i < inputs.length; i++) {
            inputs[i] = sc.nextInt();
        }

        System.out.println(solve(inputs));
    }

    private static int solve(int[] inputs) {
        int ret = 0;
        int sum = 0;

        for (int input : inputs) {
            if (input == 1) {
                sum++;
                ret += sum;
            } else {
                sum = 0;
            }
        }

        return ret;
    }

}

File: algorithm/src/main/java/baekjoon/p02506/Main.kt
package baekjoon.p02506

fun main(args: Array<String>) {
    val n = readLine()!!.toInt()
    val inputs = readLine()!!.trim().split(" ").map { it.toInt() }.toIntArray()

    val ret = solve(inputs)
    println(ret)
}

private fun solve(inputs: IntArray): Int {
    var ret = 0
    var sum = 0

    for (input in inputs) {
        if (input == 1) {
            sum++
            ret += sum
        } else {
            sum = 0
        }
    }

    return ret
}